src/python/pants/backend/adhoc/run_system_binary.py

```diff
@@ -128,14 +127,21 @@ async def _find_binary(
 
 
 @rule(level=LogLevel.DEBUG)
-async def create_system_binary_run_request(field_set: SystemBinaryFieldSet) -> RunRequest:
+async def create_system_binary_run_request(
+    field_set: SystemBinaryFieldSet,
+    system_binaries_environment: SystemBinariesSubsystem.EnvironmentAware,
```

Comment:
Should just call this `system_binaries` throughout (as you do in several other places, it's good to be consistent). Sometimes we append `_env_aware` to the arg name, but in this case there are no non-env-aware options in the subsystem, so there's no need to make the distinction.